### ISSUE TITLE
Add a Twig filter or function for the `safeTruncate` method

### ISSUE THREAD

**Issue by lindseydiloreto** [craftcms/cms#6838]
There is a function in PHP which is extremely helpful when trying to shorten long blocks of text...

```php
StringHelper::safeTruncate
```

It would be even _more_ helpful if we could access that method directly from Twig! 😄

Any thoughts on surfacing that method as a Twig filter and/or function?

```twig
{# Filter #}
{{ myLongTextBlock | safeTruncate(140, '...') }}

{# Function #}
{{ safeTruncate(myLongTextBlock, 140, '...') }}
```

**Comment by lindseydiloreto** [comment]
Ah, brilliant!! Thanks @brandonkelly! 🍺

**Comment by brandonkelly** [comment]
Just released Craft 3.5.10 with this 🎉

Went with `|truncate` instead of `|safeTruncate`, since I don’t think we would ever add a non-word-based truncate filter anyway (that’s what `|slice` is for).

Docs here: https://craftcms.com/docs/3.x/dev/filters.html#truncate

**Comment by lindseydiloreto** [comment]
I wondered about that, makes sense to me.

You'll be happy to know that I've already updated, and the new filter is working perfectly! 😄
